node-ts/src/routes/workplace.routes.ts

```diff
@@ -0,0 +1,73 @@
+﻿import { Router, Request, Response } from "express";
+import { Logger, ILogObj } from "tslog";
+import { sanitize } from "dompurify"
+import {WorkplaceBody} from "../entities/workplaceRequest";
+import {Topic} from "../enums/workplaceEnums";
+import {handleGroup} from "../handlers/workplace.handler";
+const log: Logger<ILogObj> = new Logger();
+
+const {
+    WORKPLACE_VERIFY_TOKEN,
+} = process.env
+
+const router = Router();
+
+/** Verify server in webhook configuration*/
+router.get('/', (req, res) => {
+    const { mode, challenge, verify_token } = {
+        mode: req.query["hub.mode"] ?? '',
+        challenge: req.query["hub.challenge"] ?? '',
+        verify_token: req.query["hub.verify_token"] ?? '',
+    }
+    
+    if (!(mode === 'subscribe' && verify_token === WORKPLACE_VERIFY_TOKEN)) {
+        log.error('Failed validation. Make sure the validation tokens match.')
+        res.sendStatus(403);
+        return
+    }
+    
+    log.debug('Validating webhook')
+    res.status(200).send(sanitize(challenge.toString()))
+});
+
+
+router.post('/', async (req: Request<unknown,unknown,WorkplaceBody>, res: Response) => {
+    try {
+        // verifyRequestSignature({},'')
+        switch (req.body.object) {
+            case Topic.group: 
+                await handleGroup(req.body.entry)
+                break;
+            default:
+                log.warn('Unhandled Webhook Object', req.body.object);
+        }
+    } catch (e) {
+        log.error(e);
+    } finally {
+        res.sendStatus(200);
+    }
+});
+
+/** Verify that the callback came from Facebook. */
+function verifyRequestSignature(payload:unknown,signature:unknown) {
+    if (!signature) throw new Error('No signature in header.')
+    // const signature = req.headers['x-hub-signature'];
```

Comment:
Keeping in mind that this is still a work in progress, we should still avoid commented-out code. :)